app/src/main/java/de/westnordost/streetcomplete/data/preferences/Preferences.kt

```diff
@@ -0,0 +1,246 @@
+package de.westnordost.streetcomplete.data.preferences
+
+import com.russhwolf.settings.ObservableSettings
+import com.russhwolf.settings.SettingsListener
+import com.russhwolf.settings.boolean
+import com.russhwolf.settings.float
+import com.russhwolf.settings.int
+import com.russhwolf.settings.long
+import com.russhwolf.settings.nullableString
+import com.russhwolf.settings.string
+import de.westnordost.streetcomplete.data.osm.mapdata.LatLon
+
+class Preferences(private val prefs: ObservableSettings) {
+    // application settings
+    var language: String? by prefs.nullableString(LANGUAGE_SELECT)
+
+    var theme: Theme
+        set(value) { prefs.putString(THEME_SELECT, value.name) }
+        get() {
+            val value = prefs.getStringOrNull(THEME_SELECT)
+            // AUTO setting was removed because as of June 2024, 95% of active installs from
+            // google play use an Android where AUTO is deprecated
+            return if (value == "AUTO" || value == null) Theme.SYSTEM else Theme.valueOf(value)
+        }
+
+    var autosync: Autosync
+        set(value) { prefs.putString(AUTOSYNC, value.name) }
+        get() = prefs.getStringOrNull(AUTOSYNC)?.let { Autosync.valueOf(it) } ?: DEFAULT_AUTOSYNC
+
+    var keepScreenOn: Boolean by prefs.boolean(KEEP_SCREEN_ON, false)
+
+    var resurveyIntervals: ResurveyIntervals
+        set(value) { prefs.putString(RESURVEY_INTERVALS, value.name) }
+        get() = prefs.getStringOrNull(RESURVEY_INTERVALS)?.let { ResurveyIntervals.valueOf(it) }
+            ?: DEFAULT_RESURVEY_INTERVALS
+
+    var showAllNotes: Boolean by prefs.boolean(SHOW_ALL_NOTES, false)
+
+    fun getLastPicked(key: String): List<String> =
+        prefs.getStringOrNull(LAST_PICKED_PREFIX + key)?.split(',') ?: listOf()
```

Comment:
Using comma as a separator is a bit dangerous. Cycleways are serialized as json and have two properties, so `{"cycleway":"SEPARATE","direction":"FORWARD"}` is serialized correctly, but broken on deserialization as it's now `{"cycleway":"SEPARATE"` and `"direction":"FORWARD"}`.

Reply:
Okay, but this is not a change. It has been that way also before. So how can this be an issue now and before not?

Reply:
Oh, nevermind, yes, that part didn't use the last picked values store class, only the same key

Reply:
I'm not sure, I only noticed that last picked selection is not shown in cycleway quest and overlay, and decided to have a look where things are going wrong. Since it did work before, I assumed thinks broke in this PR.